rbx_binary/src/deserializer.rs

```diff
@@ -947,6 +974,37 @@ impl<R: Read> BinaryDeserializer<R> {
                     });
                 }
             },
+            Type::SharedString => match canonical_type {
+                VariantType::SharedString => {
+                    let mut values = vec![0; type_info.referents.len()];
+                    chunk.read_interleaved_u32_array(&mut values)?;
+
+                    for (value, referent) in values.into_iter().zip(&type_info.referents) {
+                        let shared_string = self.shared_strings.get(&value).ok_or_else(|| {
+                            InnerError::InvalidPropData {
+                                type_name: type_info.type_name.clone(),
+                                prop_name: prop_name.clone(),
+                                valid_value: "a valid SharedString",
+                                actual_value: format!("{:?}", value),
+                            }
+                        })?;
+
+                        let value: BinaryString = shared_string.data().into();
```

Comment:
Why convert the value into a `BinaryString` here? We have support for `SharedString` in rbx_dom_weak, so we should use that here.